src/platform/backends/shared/qemu_img_utils/qemu_img_utils.h

```diff
@@ -19,15 +19,23 @@
 #define MULTIPASS_QEMU_IMG_UTILS_H
 
 #include <multipass/path.h>
+#include <multipass/platform.h>
+
+#include <optional>
 
 namespace multipass
 {
 class MemorySize;
+class QemuImgProcessSpec;
 
 namespace backend
 {
+Process::UPtr checked_exec_qemu_img(std::unique_ptr<QemuImgProcessSpec> spec,
+                                    const std::string& custom_error_prefix = "Internal error",
+                                    std::optional<int> timeout = std::nullopt);
```

Comment:
It's only that it expresses intent better. Having an int defaulting to 0 would have worked, but you'd need a comment to properly convey what 0 meant: not immediate timeout, not no timeout, but default timeout. In general it is better to express ideas with language constructs than with comments. It lets you enlist the compiler's help to verify that those ideas hold and make it more obvious to the reader/client.

Now, in reality this should also have a comment to explain what `nullopt` means, but I think it is more intuitive that you can abstain from choosing a timeout. Admittedly it is only marginally better in this case, but it also costs very nearly nothing (a few keystrokes).

Ideally, we should have a type that expressed custom or default timeouts, such that you could say `ExecTimeout::default` or something. Or `enum Default{Default}; std::variant<int,Default> timeout`. But that would start to feel like a bit of overkill here, since we'd be introducing a new type and we should then use it wherever we have nested defaults (nested functions with the same default parameter).

Let me know if you would like to discuss it further!

Reply:
Ok, thanks for the explanation! I was just curious if there was a functional difference between the two approaches.